packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/WorkflowService.java

```diff
@@ -1,21 +1,32 @@
 package software.uncharted.terarium.hmiserver.service.data;
 
 import org.springframework.stereotype.Service;
+
 import software.uncharted.terarium.hmiserver.configuration.Config;
 import software.uncharted.terarium.hmiserver.configuration.ElasticsearchConfiguration;
 import software.uncharted.terarium.hmiserver.models.dataservice.workflow.Workflow;
+import software.uncharted.terarium.hmiserver.repository.data.WorkflowRepository;
 import software.uncharted.terarium.hmiserver.service.elasticsearch.ElasticsearchService;
 
 @Service
-public class WorkflowService extends TerariumAssetService<Workflow>{
-
+public class WorkflowService extends TerariumAssetServiceWithES<Workflow, WorkflowRepository> {
```

Comment:
What do we expect the workflow update pattern to be like? 

- Is it such that we'd send in a workflow object and the API will automatically figure out which nodes/edges it should update? or
- would we need to uniquely update each node/edge individually?

Reply:
Currently I left it at 1 row per Workflow. The only cubmersome part is that `WorkflowEdges` embed the entire node, rather than just referencing it by id. 

@mwdchang Are workflow nodes / edges ever shared between workflows? Can we get away by just having the edge reference the target / source by id?

Reply:
The `WorkflowNode<any>['id'];` syntax confused me, I thought those were the full types :) I'll fix my code